### PR TITLE
feat(android): filter containers by category

### LINKED ISSUE #19: Filter containers by category
Refs: [Figma](https://www.figma.com/design/JvzZN6eHqoo2zgx6S6qbpT/EcoMap?node-id=260-3803&t=VfZRYYYFVdIaqASR-4)

Allow multiple filters in a horizontal scroll bar for:
- GENERAL
- PAPER
- PLASTIC
- METAL
- GLASS
- ORGANIC
- HAZARDOUS

### PR DESCRIPTION
**Pull request by Goncalo-Marques**
Refs: closes #19

## Summary

Add a chip group to allow the user to filter the containers in the map by category.

### CODE REVIEW

File: android/app/src/main/java/com/ecomap/ecomap/MainActivity.kt [hunk 1]
```diff
@@ -109,29 +109,101 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback {
...
         val buttonMyLocation: FloatingActionButton = findViewById(R.id.button_my_location)
 
         // Set button functions.
+        populateChipGroupContainerFilter(chipGroupContainerFilter)
         buttonMyLocation.setOnClickListener { focusMyLocation() }
     }
 
+    /**
+     * Populates the given chip group with all the available container categories.
+     * */
```
Comment (joaotomaspinheiro) [inline android/app/src/main/java/com/ecomap/ecomap/MainActivity.kt]:
```suggestion
     */
```

File: android/app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -18,17 +46,39 @@
         app:layout_constraintTop_toTopOf="parent"
         tools:layout="@layout/fragment_map" />
 
+    <HorizontalScrollView
+        android:id="@+id/horizontal_scroll_view_container_filter"
+        style="@style/Theme.EcoMap"
```
Comment (joaotomaspinheiro) [inline android/app/src/main/res/layout/activity_main.xml]: Is `style="@style/Theme.EcoMap"` needed?

Comment (Goncalo-Marques) [reply android/app/src/main/res/layout/activity_main.xml]: Not really, just a leftover from testing the style.
I will remove that

**Review by joaotomaspinheiro — DISMISSED** [review]

File: android/app/src/main/java/com/ecomap/ecomap/Common.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.android.volley.VolleyError
+import com.ecomap.ecomap.clients.ecomap.http.ApiClient
+import com.ecomap.ecomap.signin.SignInActivity
+
+/**
+ * Class containing common methods and attributes.
+ */
+class Common {
+    companion object {
+        fun handleVolleyError(context: Context, activity: Activity, error: VolleyError) {
```
Comment (joaotomaspinheiro) [inline android/app/src/main/java/com/ecomap/ecomap/Common.kt:24]: Add documentation to this method.

**Review by joaotomaspinheiro — APPROVED** [review]

**Pull request merged**